superset-frontend/src/explore/components/ExploreViewContainer/index.jsx

```diff
@@ -460,15 +460,21 @@ function ExploreViewContainer(props) {
 
   const chartIsStale = useMemo(() => {
     if (lastQueriedControls) {
-      const changedControlKeys = Object.keys(props.controls).filter(
-        key =>
-          typeof lastQueriedControls[key] !== 'undefined' &&
-          !areObjectsEqual(
-            props.controls[key].value,
-            lastQueriedControls[key].value,
-            { ignoreFields: ['datasourceWarning'] },
-          ),
-      );
+      const { controls } = props;
+      const changedControlKeys = Object.keys(controls).filter(key => {
```

Comment:
This change fixes a problem with the algorithm that detect a stale chart. Previously, if the values being compared were not objects, we had wrong results. For example, if a user changed the row limit, which is a number, the banner indicating a staled chart was not shown. This should significantly improve user experience when changing control values.